DedicatedHostsManager/DedicatedHostEngine/DedicatedHostEngine.cs

```diff
@@ -826,5 +969,92 @@ await Policy
         }
+
+        private IList<(int fd, int numberOfHosts)> CalculatePlatformFaultDomainToHost(
+            DedicatedHostGroup dhGroup,
+            IList<DedicatedHost> existingHostsOnDHGroup,
+            string vmSku,
+            int vmInstancesRequested,
+            int vmCapacityPerHost,
+            int? platformFaultDomain)
+        {
+            var dedicatedHosts = new List<(int fd, int numberOfHosts)>();
+            var platformFaultDomains = DetermineFaultDomainsForPlacement(dhGroup.PlatformFaultDomainCount, platformFaultDomain);
+            var vmsRequiredPerFaultDomain = (int)Math.Round((decimal)vmInstancesRequested / platformFaultDomains.Length, MidpointRounding.ToPositiveInfinity);
+
+            foreach (var fd in platformFaultDomains)
+            {
+                var dhInFaultDomain = existingHostsOnDHGroup.Where(c => c.PlatformFaultDomain == fd);
+                var availableVMCapacityInFaultDomain = 0;
+                foreach (var host in dhInFaultDomain)
+                {
+                    // Existing hosts can be different sku then DH Sku determined for VM size.
+                    availableVMCapacityInFaultDomain += (int)(host.InstanceView?.AvailableCapacity?.AllocatableVMs?
+                        .FirstOrDefault(v => v.VmSize.Equals(vmSku, StringComparison.InvariantCultureIgnoreCase))?.Count ?? 0);
+                }
+
+                if (vmsRequiredPerFaultDomain > availableVMCapacityInFaultDomain)
+                {
+                    dedicatedHosts.Add((fd, (int)(Math.Round(((decimal)vmsRequiredPerFaultDomain - availableVMCapacityInFaultDomain) / vmCapacityPerHost, MidpointRounding.ToPositiveInfinity))));
+                }
+            }
+
+            return dedicatedHosts;
+
+            int[] DetermineFaultDomainsForPlacement(int dhGroupFaultDomainCount, int? platformFaultDomain)
+            {
+                if (platformFaultDomain != null && platformFaultDomain > (dhGroupFaultDomainCount - 1))
+                {
+                    throw new Exception($"Invalid requested Platform Fault domain -Dedicated Host Group Fault Domains = {dhGroupFaultDomainCount}, requested Platform Fault domain = {platformFaultDomain}");
+                }
+
+                return platformFaultDomain switch
+                {
+                    0 => new int[] { 0 },
+                    1 => new int[] { 1 },
+                    2 => new int[] { 2 },
+                    _ => Enumerable.Range(0, dhGroupFaultDomainCount).ToArray() // As # of small, perf of using Range not significant
+                };
+            }
+        }
+
+        private (string dhSku, int vmCapacity) GetVmCapacityPerHost(string location, string vmSku)
+        {
+            var matchingConfig = _config.DedicatedHostConfigurationTable
+                .Where(c => (c.Location == "default" || c.Location.Equals(location, StringComparison.OrdinalIgnoreCase))
```

Comment:
What's the purpose of the "default" check?

Reply:
"DedicatedHostMapping" configuration defines mapping between host SKU, vm family, vm size. Some regions might have different capacity considerations and in such use case, region specific settings would be provided. If none provided, "default" settings would be applied. Requirement specified this as sample
vm_family_to_host_sku= {"DSv3":{"Default":"DSv3-Type1", "ES East":"DSv3-Type2"},
                        "ESv3":{"Default":"ESv3-Type1", "ES East":"ESv3-Type2"},…} 

Currently ONLY default defined in configuration sample provided for "DSv3-Type1" as it is standard across all regions.